core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3449,6 +3449,8 @@ class KafkaApis(val requestChannel: RequestChannel,
 
   def handleInitializeShareGroupStateRequest(request: RequestChannel.Request): CompletableFuture[Unit] = {
     val initializeShareGroupStateRequest = request.body[InitializeShareGroupStateRequest]
+    // We do not need a check for isShareGroupProtocolEnabled in this RPC since there is a check for it in ShareFetch/ShareAcknowledge RPCs,
+    // hence requests won't reach Persister. We might need to hit this API for debugging purposes.
```

Comment:
The later section seems not required. I ll let @smjn decide what's best.
```suggestion
    // hence requests won't reach Persister.
```

Reply:
Sounds good, lets remove it.